tests/console/kubeadm.pm

```diff
@@ -19,24 +19,20 @@ use utils "systemctl";
 sub run {
     select_console("root-console");
 
-    record_info 'etcd', 'Stop etcd and clean up';
-    systemctl 'disable --now etcd';
-    script_run 'rm -r /var/lib/etcd/*';
-
-    record_info 'Setup', 'Test: Package Installation';
-    my $packages = 'kubernetes-client kubernetes-kubelet kubernetes-kubeadm docker-kubic cri-tools';
-    trup_install($packages);
+    record_info 'Test #1', 'Test: Initialize kubeadm';
+    assert_script_run('kubeadm init --cri-socket=/var/run/crio/crio.sock --pod-network-cidr=10.244.0.0/16', 180);
 
-    record_info 'Prepare', 'Test: Enable and start required services';
-    systemctl('enable --now kubelet docker');
-    systemctl('is-active docker');
-    systemctl('is-active kubelet');
+    record_info 'Test #2', 'Test: Configure kubectl';
+    assert_script_run('mkdir -p ~/.kube');
+    assert_script_run('cp -i /etc/kubernetes/admin.conf ~/.kube/config');
 
-    record_info 'Test #1', 'Test: Initialize kubeadm';
-    record_soft_failure "bsc#1093132" if script_run('kubeadm init');
-    script_run('kubeadm reset');
-    assert_script_run('kubeadm init', 180);
+    record_info 'Test #3', 'Test: Configure flannel';
+    assert_script_run('kubectl apply -f https://raw.githubusercontent.com/coreos/flannel/bc79dd1505b0c8681ece4de4c0d86c5cd2643275/Documentation/kube-flannel.yml', 180);
```

Comment:
What about using `https://raw.githubusercontent.com/coreos/flannel/master/Documentation/kube-flannel.yml` instead? I also, the `180` it's not needed I think. What `kubectl apply` does takes one or two seconds. You need to make sure though that this pod is up and running, so I would add an extra check for this, something like: `kubectl get pods --all-namespaces | grep coredns | grep Running | wc -l | grep 2`

Reply:
Because upstream say to use that path, not master

Reply:
master is a changing thing not guaranteed to be stable tested. That would be worst than using the known-tested fix-commit version.